src/main/resources/db/migration/V1_95_2__referral_details_unique_index.sql

```diff
@@ -0,0 +1 @@
+CREATE UNIQUE INDEX CONCURRENTLY IF NOT EXISTS referral_details_superseded_ix ON referral_details (superseded_by_id) WHERE superseded_by_id IS NOT NULL;
```

Comment:
could we also add an index on `referral_detail(referral_id)` please (primary lookup mechanism)?